classifiers/reference_quality/chunked_sentence_complexity/__init__.py

```diff
@@ -0,0 +1,79 @@
+from pydantic import BaseModel
+from typing import Optional
+from extractors.util.spacy import SpacySingleton
+from collections import Counter
+import textstat
+
+INPUT_EXAMPLE = {
+    "text": "Wow, this is really cool!", 
+    "language": "en",
+    "spacy_model": "en_core_web_sm"
+}
+
+class ChunkedSentenceComplexityModel(BaseModel):
+    text: str
+    language: Optional[str]
+    spacy_model: Optional[str]
+
+    class Config:
+        schema_extra = {"example": INPUT_EXAMPLE}
+
+
+def setall(d, keys, value):
+    for k in keys:
+        d[k] = value
+
+OUTCOMES = {}
+setall(OUTCOMES, range(90, 122), "very easy")
+setall(OUTCOMES, range(80, 90), "easy")
+setall(OUTCOMES, range(70, 80), "fairly easy")
+setall(OUTCOMES, range(60, 70), "standard")
+setall(OUTCOMES, range(50, 60), "fairly difficult")
+setall(OUTCOMES, range(30, 50), "difficult")
+setall(OUTCOMES, range(0, 30), "very difficult")
+
+
+def get_mapping_complexity(score):
+    if score < 0:
+        return OUTCOMES[0]
+    return OUTCOMES[int(score)]
+
+
+def sentence_complexity(text):
+    sentence_complexity_score = textstat.flesch_reading_ease(text)
+    sentence_complexity = get_mapping_complexity(sentence_complexity_score)
+    return sentence_complexity
+    
+    
+def chunked_sentence_complexity(req: ChunkedSentenceComplexityModel):
```

Comment:
not sure if the approach is optimal since we map and then merge and average of the mapped value. 
Imo a better approach would be to sum sentence complexity scores (0-122), average them and then use that to define a final complexity. 

With the current approach a 5x easy (80) + 1x standard(60) + 1x very difficult(0):

would result in `(2*5 + 4 + 7) / 7 = 3 =>fairly easy` . 

with `(80*5 + 60 + 0) / 7 = ~72 => standard`


This would also improve the performance since we don't store / use string values to calculate, only the numbers itself.

example function further below

Reply:
Great suggestion Jens. Yes this approach is much more lightweight. I have implemented it.